pkg/status/health/health.go

```diff
@@ -0,0 +1,141 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package health
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/cihub/seelog"
+)
+
+// DefaultPingFreq holds the preferred time between two pings
+const DefaultPingFreq = 15 * time.Second
+
+// DefaultTimeout holds the duration used for default (twice DefaultPingFreq)
+const DefaultTimeout = 30 * time.Second
+
+// ID objects are returned when registering and are to be used when pinging
+type ID string
+
+// Status represents the current status of registered components
+type Status struct {
```

Comment:
Why don't you store the health directly as a bool in the component?

Reply:
This type is only built by GetStatus. Using a well-defined type helps with the marshalling/unmarshalling